### PR TITLE
[FIX]: wrange-loop-contruct

### LINKED ISSUE #5929: warning: loop variable ‘kv’ of `some` type binds to a temporary constructed from type `some`
```shell
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp: In member function ‘void OpenMS::MRMFeatureFilter::FilterFeatureMap(OpenMS::FeatureMap&, const OpenMS::MRMFeatureQC&, const OpenMS::TargetedExperiment&)’:
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:205:70: warning: loop variable ‘kv’ of type ‘const std::pair<OpenMS::String, std::pair<double, double> >&’ binds to a temporary constructed from type ‘const std::pair<const OpenMS::String, std::pair<double, double> >’ [-Wrange-loop-construct]
  205 |             for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
      |                                                                      ^~
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:205:70: note: use non-reference type ‘const std::pair<OpenMS::String, std::pair<double, double> >’ to make the copy explicit or ‘const std::pair<const OpenMS::String, std::pair<double, double> >&’ to prevent copying
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp: In member function ‘void OpenMS::MRMFeatureFilter::FilterFeatureMapPercRSD(OpenMS::FeatureMap&, const OpenMS::MRMFeatureQC&, const OpenMS::MRMFeatureQC&)’:
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:399:70: warning: loop variable ‘kv’ of type ‘const std::pair<OpenMS::String, std::pair<double, double> >&’ binds to a temporary constructed from type ‘const std::pair<const OpenMS::String, std::pair<double, double> >’ [-Wrange-loop-construct]
  399 |             for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
      |                                                                      ^~
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:399:70: note: use non-reference type ‘const std::pair<OpenMS::String, std::pair<double, double> >’ to make the copy explicit or ‘const std::pair<const OpenMS::String, std::pair<double, double> >&’ to prevent copying
```

### PR DESCRIPTION
**Pull request by Amit0617**
## Description

Fixes #5929
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by timosachsenberg — APPROVED** [review]
Thanks!

File: src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp [hunk 1]
```diff
@@ -202,7 +202,7 @@ namespace OpenMS
               }
             }
 
-            for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
+            for (const std::pair<const String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
```
Comment (cbielow) [inline src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:205]: I think `auto` would have been the better choice here
```
for (const auto& kv : ...
```

But there will be a new PR coming up, which fixes a few other things (mostly naming of things as `it` which are not iterators), so just for future reference :)

**Comment by cbielow** [comment]
just as an idea for the future workflow:
IMHO creating a new issue, just to go ahead and fix it immediately in a PR which links to it, creates a lot of "noise", and when it distributes the problem (compiler warning/output) and the fix (PR) across two different pages, which actually takes more effort to review it.
Why not simply open a PR which contains the fix and has the compiler warning in its description? All information is in one place, there are less emails flying around in everybodys inbox and it saves a bit of time for everybody.

**Comment by cbielow** [comment]
If you keep up the old workflow, you can open hundreds of issues (yes, we have that many compiler warnings)... Its just a bit too much, and it distracts from the higher level issues -- at least in my case.

**Comment by Amit0617** [comment]
> just as an idea for the future workflow: IMHO creating a new issue, just to go ahead and fix it immediately in a PR which links to it, creates a lot of "noise", and it distributes the problem (compiler warning/output) and the fix (PR) across two different pages, which actually takes more effort to review it. Why not simply open a PR which contains the fix and has the compiler warning in its description? All information is in one place, there are less emails flying around in everybodys inbox and it saves a bit of time for everybody.

Ok will follow that one in future
but it had started this way that, I wasn't sure if i will really be fixing it or will be able to do that(knew nothing about Qt). So i just created issues for bringing attention towards it.

**Comment by Amit0617** [comment]
> If you keep up the old workflow, you can open hundreds of issues (yes, we have that many compiler warnings)... Its just a bit too much, and it distracts from the higher level issues -- at least in my case.

I don't think so atleast now there are just few more. (atleast for my gcc version, if that affects)

**Pull request merged**